### PR TITLE
docs: fixed broken links and adopted htmltest - fixes #3834

### LINKED ISSUE #3834: [BUG] broken links in the docs
The current link checker used for docs fails to catch broken links.

`htmltest` has better results:

```
htmltest started at 02:51:26 on docs/build
========================================================================
crc/getting_started/getting_started/using/index.html
  Non-OK status: 404 --- crc/getting_started/getting_started/using/index.html --> https://docs.openshift.com/container-platform/latest/cli_reference/developer_cli_odo/installing-odo.html
  Non-OK status: 404 --- crc/getting_started/getting_started/using/index.html --> https://docs.openshift.com/container-platform/latest/cli_reference/developer_cli_odo/understanding-odo.html
crc/404.html
  target does not exist --- crc/404.html --> /getting_started/getting_started/introducing/
========================================================================
✘✘✘ failed in 1.130990399s
3 errors in 10 documents
```

### PR DESCRIPTION
**Pull request by themr0c**
docs: fixed broken links and adopted htmltest
chore: use the standard `httpd` container image to serve the docs locally

fixes #3834

```
make build_docs docs_check_links
podman run -v /home/ffloreth/src/gh/themr0c/crc:/antora:Z --rm docker.io/antora/antora:latest antora-playbook.yml
podman run -v /home/ffloreth/src/gh/themr0c/crc:/test:Z --rm docker.io/wjdp/htmltest:latest -c .htmltest.yml
htmltest started at 01:15:50 on docs/build
========================================================================
✔✔✔ passed in 1.345655164s
tested 10 documents
```

### CODE REVIEW

File: antora-playbook.yml [hunk 1]
```diff
@@ -1,69 +1,68 @@
...
+    crc-download-url: https://console.redhat.com/openshift/create/local
+    crc-gsg-url: https://crc.dev/crc/
+    openshift-installer-url: https://console.redhat.com/openshift/create
+    openshift-docs-url: https://docs.openshift.com/container-platform/latest
+    openshift-docs-url-landing-page: "{openshift-docs-url}/welcome/index.html#developer-activities"
+    oc-download-url: https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/
+    odo-docs-url: https://odo.dev/docs/introduction
+    odo-docs-url-installing: https://odo.dev/docs/overview/installation
+    telemetry-notice-url: https://developers.redhat.com/article/tool-data-collection
+    rhel-resolved-docs: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/using-different-dns-servers-for-different-domains_configuring-and-managing-networking
```
Comment (praveenkumar) [inline antora-playbook.yml:68]: These space related changes shouldn't be part of the commit.

**Comment by themr0c** [comment]
Rebased and squashed.

**Comment by themr0c** [comment]
Fixed the features lost in rebase.
